### ISSUE TITLE
AttributeError: module 'avro.io' has no attribute 'AvroTypeException'

### ISSUE THREAD

**Issue by mishbahr** [linkedin/datahub#4050]
**Describe the bug**
Writing custom ingestion for my argo based workflow.

```
Traceback (most recent call last):
  File "../datahub-emit/lineage.py", line 32, in <module>
    emitter.emit_mce(mce)
  File "../datahub-emit/env/lib/python3.9/site-packages/datahub/emitter/rest_emitter.py", line 167, in emit_mce
    raw_mce_obj = mce.proposedSnapshot.to_obj()
  File "/Users/mishbah/Workspace/datahub-emit/env/lib/python3.9/site-packages/avrogen/dict_wrapper.py", line 45, in to_obj
    return conv.to_json_object(self, self.RECORD_SCHEMA)
  File "../datahub-emit/env/lib/python3.9/site-packages/avrogen/avrojson.py", line 145, in to_json_object
    raise io.AvroTypeException(writers_schema, data_obj)
AttributeError: module 'avro.io' has no attribute 'AvroTypeException'
```

**To Reproduce**

```
# lineage.py

from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
from datahub.emitter.rest_emitter import DatahubRestEmitter
from datahub.emitter import mce_builder

from datahub.metadata.schema_classes import (
    MetadataChangeEventClass,
    DataFlowSnapshotClass,
    DataFlowInfoClass,
)


emitter = DatahubRestEmitter("http://localhost:8080")
emitter.test_connection()

dataflow = DataFlowSnapshotClass(
    urn=mce_builder.make_data_flow_urn(
        orchestrator="argo", flow_id="42", cluster="DEV"
    ),
    aspects=[
        DataFlowInfoClass(
            name="hello_datahub",
            description="Hello Datahub",
            externalUrl="http://example.com",
            customProperties={"x": 1},
        )
    ],
)

mce = MetadataChangeEventClass(proposedSnapshot=dataflow)

emitter.emit_mce(mce)
```
```
$ python3 lineage.py
```

Package List

```
acryl-datahub==0.8.24.1
avro==1.11.0
avro-gen3==0.7.1
certifi==2021.10.8
charset-normalizer==2.0.11
click==8.0.3
click-default-group==1.2.2
docker==5.0.3
entrypoints==0.4
expandvars==0.7.0
idna==3.3
mypy-extensions==0.4.3
progressbar2==4.0.0
pydantic==1.9.0
python-dateutil==2.8.2
python-utils==3.1.0
pytz==2021.3
pytz-deprecation-shim==0.1.0.post0
PyYAML==6.0
requests==2.27.1
six==1.16.0
stackprinter==0.2.5
tabulate==0.8.9
toml==0.10.2
typing-extensions==3.10.0.2
typing-inspect==0.7.1
tzdata==2021.5
tzlocal==4.1
urllib3==1.26.8
websocket-client==1.2.3
```

**Additional context**
Looks like the error is only raised when `customProperties` includes a non-string value i.e `{"x": "1"}` works fine.

**Comment by RyanHolstien** [comment]
Hi! So there are a couple of things going on here:

1) customProperties is a dict(str, str) so it only accepts a map of string -> string. This aligns with our Avro type. This is not a bug.

2) In avro.io 1.10, AvroTypeException got moved to a different package. We use a library that generates the code that throws that exception and it is generating the conversion method with an older version of avro.io than 1.11 that has the exception type in the old locaiton. Because Python is interpreted, it only hits the definition error when it actually attempts to throw that error. We'll need to constrain the dependencies tighter here (or loosen the one for the generator library) so that the type def error does not occur in cases where there is an AvroTypeException.

**Comment by shirshanka** [comment]
Fixed in `acryl-datahub==0.8.24.2` (https://github.com/acryldata/datahub/releases/tag/v0.8.24.2)
